Back-End-Node/server.js

```diff
@@ -1,23 +1,126 @@
 const express = require("express");
 const app = express();
 const { pool } = require("./dbConfig");
-
+var http = require("http");
+var fs = require("fs");
+var https = require("https");
+var fetch = require("node-fetch");
 app.use(express.json());
 require("dotenv").config();
 
+var options = {
+  key: fs.readFileSync("client-key.pem"),
+  cert: fs.readFileSync("client-cert.pem"),
+};
+// app.use(function (req, res, next) {
+//   res.writeHead(200);
+//   res.end("hello world\n");
+//   next();
+// });
+
+const { WebClient } = require("@slack/web-api");
+const client = new WebClient();
+const client_id = "6209798254180.6230482777808";
+const client_secret = "4fdc85a575bd64126a04a533934d5aa7";
+app.get("/auth/redirect", async (req, res) => {
+  // TODO: verify state parameter
+  try {
+    const query = {
+      headers: { "Content-Type": "application/x-www-form-urlencoded" },
+      method: "POST",
+      body: `code=${req.query.code}&client_id=${client_id}&client_secret=${client_secret}`,
+    };
+    const authResponse = await fetch(
+      "https://slack.com/api/oauth.v2.access",
+      query
+    );
+    const authJson = await authResponse.json();
+
+    console.log("authJson", authJson);
+
+    const userDataResponse = await fetch(
+      `https://slack.com/api/users.identity?user=${authJson.authed_user.id}`,
+      {
+        headers: {
+          Authorization: `Bearer ${authJson.authed_user.access_token}`,
+        },
+      }
+    );
+    const userDataJson = await userDataResponse.json();
+   
+    console.log("userDataJson", userDataJson);
+
+    res.redirect("http://localhost:3000/oauthdone?code=1234");
```

Comment:
will localhost work here?

Reply:
ofcourse it is not working we need to create that endpoint. It is just an example to replace it with real one.